edlib/src/edlib.cpp

```diff
@@ -1419,16 +1418,17 @@ static int obtainAlignmentHirschberg(
  */
 static string transformSequences(const char* const queryOriginal, const int queryLength,
                                  const char* const targetOriginal, const int targetLength,
-                                 unsigned char** const queryTransformed,
-                                 unsigned char** const targetTransformed) {
+                                 unsigned char** const queryTransformed_,
+                                 unsigned char** const targetTransformed_) {
     // Alphabet is constructed from letters that are present in sequences.
     // Each letter is assigned an ordinal number, starting from 0 up to alphabetLength - 1,
     // and new query and target are created in which letters are replaced with their ordinal numbers.
     // This query and target are used in all the calculations later.
-    *queryTransformed = static_cast<unsigned char *>(malloc(sizeof(unsigned char) * queryLength));
-    *targetTransformed = static_cast<unsigned char *>(malloc(sizeof(unsigned char) * targetLength));
+    unsigned char *queryTransformed = static_cast<unsigned char *>(malloc(sizeof(unsigned char) * queryLength));
```

Comment:
What is the benefit of introducing these "intermediary" variables, and then setting `queryTransformed_` and `targetTransformed_` only at the very end?

Reply:
As noted in 1e4d6ec0b85e6142b9740d03056ad18f4cf90050, this stops the compiler from assuming that the output pointers might alias with other pointers. In practical terms it's a micro-micro-optimisation which can be seen in this relatively noisy diff https://godbolt.org/z/YEPE7d9PK where there's one less `mov` (ie no double lookup) on the lines `queryTransformed[i] = letterIdx[c];` and `targetTransformed[i] = letterIdx[c];` (which would be `(*queryTransformed_)[i] = letterIdx[c];` and `(*targetTransformed_)[i] = letterIdx[c];` respectively).

Reply:
Got it! Pretty cool. I haven't seen the commit messages, my bad.